### PR TITLE
Feature/mds 3629 site properties for now and ada

### PR DESCRIPTION
**Pull request by mhuseinov**
- Added site properties on NOW and ADA level,
- add validation message on issue permit tab,
- exemption fee status code on issue permit, 
- fix FE bug on permit status update,
- fix for close button on edit site properties on the permit level,
- fix for commodity not showing on mines list view [(MDS-3567)](https://bcmines.atlassian.net/browse/MDS-3567),
- added FE validation on the draft permit form
image
image
image

### CODE REVIEW

File: services/core-api/app/api/mines/permits/permit_amendment/models/permit_amendment.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ class PermitAmendment(SoftDeleteMixin, AuditMixin, Base):
     def issuing_inspector_name(self):
         title = "Inspector of Mines"
 
-        #with i had null propogation
+        #with i had null propagation
```
Comment (exzizt) [inline services/core-api/app/api/mines/permits/permit_amendment/models/permit_amendment.py:88]: I don't understand this comment. What does it mean?

File: services/core-api/app/api/mines/permits/permit_amendment/resources/permit_amendment.py [hunk 2]
```diff
@@ -220,6 +222,26 @@ def create_standard_conditions(application_identity):
 
                 db.session.commit()
 
+                # create site properties if DFT permit_amendment
+                if not application_identity.now_application.site_property:
+                    site_property = MineType.find_by_permit_guid(permit_guid, mine_guid)
+
+                    if site_property:
+                        MineType.crete_or_update_mine_type_with_details(
```
Comment (exzizt) [inline services/core-api/app/api/mines/permits/permit_amendment/resources/permit_amendment.py:230]: Typo in function name "crete"

File: services/core-api/app/api/now_applications/resources/now_application_status_resource.py [hunk 3]
```diff
@@ -141,10 +161,75 @@ def put(self, application_guid):
...
+                permit_site_property = MineType.find_by_permit_guid(
+                    permit.permit_guid, now_application_identity.mine.mine_guid)
+
+                if not permit_site_property:
+                    # create site property that is linked to a permit
+                    permit_site_property = create_site_property_for_permit(
+                        now_application_identity.mine_guid, permit.permit_guid, now_site_property)
+                    permit_site_property.save()
+                else:
+                    # TODO maybe compare if anything changed
```
Comment (exzizt) [inline services/core-api/app/api/now_applications/resources/now_application_status_resource.py:208]: Is this TODO still needed?

**Review by exzizt — COMMENTED** [review]
Great work! Just a couple of minor comments

File: services/core-api/app/api/mines/permits/permit_amendment/models/permit_amendment.py (hunk 1 above)
Comment (mhuseinov) [reply services/core-api/app/api/mines/permits/permit_amendment/models/permit_amendment.py:88]: Have no idea, there was a typo that I fixed

File: services/core-api/app/api/mines/permits/permit_amendment/resources/permit_amendment.py (hunk 2 above)
Comment (mhuseinov) [reply services/core-api/app/api/mines/permits/permit_amendment/resources/permit_amendment.py:230]: done

**Review by exzizt — APPROVED** [review]
☑️ ☑️ ☑️

**Pull request merged**